great_expectations/checkpoint/actions.py

```diff
@@ -143,18 +142,31 @@ class SlackNotificationAction(ValidationAction):
           # or pass in as environment variable
           # use slack_webhook when not using slack bot token
           slack_webhook: ${validation_notification_slack_webhook}
-          # pass slack token and slack channel when not using slack_webhook
-          slack_token: # token from slack app
-          slack_channel: # slack channel that messages should go to
-          notify_on: all # possible values: "all", "failure", "success"
-          notify_with: # optional list of DataDocs site names to display in Slack message. Defaults to showing all
+          slack_token:
+          slack_channel:
+          notify_on: all
+          notify_with:
           renderer:
             # the class that implements the message to be sent
             # this is the default implementation, but you can
             # implement a custom one
             module_name: great_expectations.render.renderer.slack_renderer
             class_name: SlackRenderer
-          show_failed_expectations: *Optional* (boolean) shows a list of failed expectation types. default is false.
+          show_failed_expectations: True
+
+    Args:
+        data_context: Data Context that is used by the Action.
+        renderer (dict): Specifies the Renderer used to generate a query consumable by Slack API, e.g.:
+            {
+               "module_name": "great_expectations.render.renderer.slack_renderer",
+               "class_name": "SlackRenderer",
+           }
+        slack_webhook (str): Optional. The incoming Slack webhook to which to send notification.
+        slack_token (str): Optional. Token from Slack app. Used when not using slack_webhook.
+        slack_channel (str): Optional. Slack channel to receive notification. Used when not using slack_webhook.
+        notify_on (str): Specifies validation status that triggers notification. One of "all", "failure", "success".
+        notify_with (list[str]): Optional list of DataDocs site names to display  in Slack messages. Defaults to all.
+        show_failed_expectations (bool): Shows a list of failed expectation types.
```

Comment:
Oooh - bonus points if you can add these type annotations instead to the `__init__` method signature.